### ISSUE TITLE
Provide move/resize helpers

### ISSUE THREAD

**Issue by DemiMarie** [swaywm/wlroots#3236]
It would be nice if wlroots provided a reasonable default implementation.

**Comment by bl4ckb0ne** [comment]
You can take inspiration from [tinywl](https://github.com/swaywm/wlroots/blob/master/tinywl/tinywl.c#L359)

**Comment by DemiMarie** [comment]
> You can take inspiration from [tinywl](https://github.com/swaywm/wlroots/blob/master/tinywl/tinywl.c#L359)

That’s where the idea came from :smile:.

**Comment by bl4ckb0ne** [comment]
I'd be against a provided default, there's a few compositor out there who willingly dont implement resize/move (cage I think, and I got one too).

**Comment by DemiMarie** [comment]
> I'd be against a provided default, there's a few compositor out there who willingly dont implement resize/move (cage I think, and I got one too).

What about a `wlr_enable_auto_resize` function?

**Comment by tiosgz** [comment]
Anyway, how would it work? Wlroots doesn’t know about the compositor’s positioning system, so only bottom&right edge resizing could be implemented. Or am i missing something?

**Comment by emersion** [comment]
Maybe via grabs (https://github.com/swaywm/wlroots/issues/1506), but indeed I don't think wlroots knows enough about the compositor state to implement this. In particular, the position of toplevels is completely unknown to wlroots.

**Comment by DemiMarie** [comment]
> Anyway, how would it work? Wlroots doesn’t know about the compositor’s positioning system, so only bottom&right edge resizing could be implemented. Or am i missing something?



> Maybe via grabs (#1506), but indeed I don't think wlroots knows enough about the compositor state to implement this. In particular, the position of toplevels is completely unknown to wlroots.

True, but applications could provide callbacks to give wlroots the needed information.

**Comment by bl4ckb0ne** [comment]
I don't think giving more knowledge to wlroots is a good move, it brings a lot of edge cases.

**Comment by ammen99** [comment]
I also feel that this isn't a good fit for wlroots, current APIs are all much lower-level than this.
